content/blog/monkey-patch.md

```diff
@@ -0,0 +1,62 @@
++++
+title = "Monkey Patching"
+slug = "monkey-patch"
+date = 2023-10-17T18:29:54+05:30
+image = "/images/2023/10/monkeywithsword.jpeg"
+draft = false
+authors = ["Nitin Rajkumar"]
+description = "Monkey patching and it's sideeffects in Ruby"
+tags = ["Software Craftsmanship", "GraphQL"]
+categories = ["Software Craftsmanship", "Ruby", "Monkey Patching"]
+type = ""
++++
+
+Ruby, being a dynamic programming language provides the ability and hence freedom to reopen the existing classes and change their behavior during runtime by adding new methods or modifying existing methods. 
+
+#### How to monkey patch
+
+```ruby
+class Sound
+  def car
+		"honk"
+	end
+end
+class Sound
+	def bike
+		"peep-peep"
+	end
+end
+noise = Sound.new
+noise.car #=> honk
+noise.bike #=> peep-peep
+```
+The above is a simple example where we reopened Sound class and added a new method. It provides many other opportunities including overriding the method of the original class.
```

Comment:
Would it add value to mention that the Sound class lre existed in a library or gem and then present these methods as patches?

Reply:
We can also reopen the classes from our own code base. The other example in this blog mentions overriding String class.

Reply:
It might make it convoluted.

Reply:
We should mention the Sound class is a pre existing one, right now it will seem to be an entirely new class and will not convey monkey patching, the String.upcase example makes sense immediately